### PR TITLE
Add sqlite support

### LINKED ISSUE #3: Support sqlite
For localized `asynq` instances, it would be preferable to utilize a bare `sqlite` instance for the underlying DB, rather than a full `rqlite`. This may or may not require further modifications to the current `rqlite` support

### PR DESCRIPTION
**Pull request by elv-gilles**
support for sqlite (local) DB. Also support in-memory DB for testing purposes. 

By default the test script uses it:

```
asynq $ ./run_all_tests.sh -redis -rqlite -sqlite 
running regular tests
ok  	github.com/hibiken/asynq/internal/base	0.751s
ok  	github.com/hibiken/asynq/internal/errors	0.197s
ok  	github.com/hibiken/asynq/internal/log	0.200s

ALL TESTS SUCCEEDED

running redis tests
ok  	github.com/hibiken/asynq/internal/rdb	1.550s
ok  	github.com/hibiken/asynq	116.269s

ALL TESTS SUCCEEDED

running rqlite tests
ok  	github.com/hibiken/asynq/internal/rqlite	74.669s
ok  	github.com/hibiken/asynq	362.631s

ALL TESTS SUCCEEDED

running sqlite tests
ok  	github.com/hibiken/asynq/internal/rqlite	12.533s
ok  	github.com/hibiken/asynq	184.828s

ALL TESTS SUCCEEDED
```

With regular file based DB:

```
asynq $ ./run_all_tests.sh -sqlite -no-sqlite-in-memory 
running regular tests
ok  	github.com/hibiken/asynq/internal/base	0.753s
ok  	github.com/hibiken/asynq/internal/errors	0.194s
ok  	github.com/hibiken/asynq/internal/log	0.240s

ALL TESTS SUCCEEDED

running sqlite tests
ok  	github.com/hibiken/asynq/internal/rqlite	14.752s
ok  	github.com/hibiken/asynq	261.886s

ALL TESTS SUCCEEDED
```

### CODE REVIEW

File: asynq_test.go [hunk 1]
```diff
@@ -23,31 +27,35 @@ import (
...
+	sqliteDbTemp   bool
+	brokerType     string // redis | rqlite | sqlite
+	redisAddr      string
+	redisDB        int
 
 	useRedisCluster   bool
 	redisClusterAddrs string // comma-separated list of host:port
 
-	rqliteConfig RqliteConfig
+	rqliteConfig = (&RqliteConfig{}).InitDefaults()
```
Comment (jslching) [inline asynq_test.go:39]: This is also done line 54 it looks like

File: internal/rqlite/conn.go [hunk 2]
```diff
@@ -2,26 +2,60 @@ package rqlite
...
 }
 
 func (conn *Connection) ctx() context.Context {
 	return context.Background()
 }
+
+func Statement(sql string, params ...interface{}) *sqlite3.Statement {
+	ret := sqlite3.NewStatement(sql, params...)
+	//if len(ret.Warning) > 0 {
+	//	panic(ret.Warning)
```
Comment (jslching) [inline internal/rqlite/conn.go]: What was this for?

File: internal/sqlite/conn_sqlite.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/hibiken/asynq/internal/sqlite3"
+	"github.com/hibiken/asynq/internal/sqlite3/db"
+)
+
+func NewSQLiteConnection(ctx context.Context, dbPath string, inMemory bool) (*sqlite3.SQLiteConnection, error) {
+	op := errors.Op("NewSQLiteConnection")
+
+	var err error
+	var conn *db.DB
+	switch inMemory {
```
Comment (jslching) [inline internal/sqlite/conn_sqlite.go:18]: Simpler to just use `if` `else` statement?

File: internal/sqlite3/connection.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+type SQLiteConnection struct {
+	db *db.DB
+}
+
+func NewSQLiteConnection(db *db.DB /*, config *Config*/) *SQLiteConnection {
+	ret := &SQLiteConnection{
+		db: db,
+		/*config:       config,*/
+	}
+	/*ret.buildTables()*/
```
Comment (jslching) [inline internal/sqlite3/connection.go]: Still needed?

File: tools/asynq/cmd/root.go [hunk 5]
```diff
@@ -38,7 +38,7 @@ var (
 	clusterAddrs    string
 	tlsServerName   string
 
-	rqliteConfig asynq.RqliteConfig
+	rqliteConfig = (&asynq.RqliteConfig{}).InitDefaults()
```
Comment (jslching) [inline tools/asynq/cmd/root.go:41]: Same as line 89?

**Review by jslching — APPROVED** [review]
LGTM, thanks!

I did not review most of the `internal/sqlite3` additions since they seem to be just clones of another project

**Review by lukseven — APPROVED** [review]

File: internal/rqlite/conn.go (hunk 2 above)
Comment (elv-gilles) [reply internal/rqlite/conn.go]: > What was this for?

The returned `Statement` contained a warning if the count of `?` in the sql string does not match the count of params. I enabled the panic when coding.

I replaced the warning made at construction by a function one can call and added comments on `sqlite3.Statement`.

**Comment by elv-gilles** [comment]
> most of the `internal/sqlite3` additions since they seem to be just clones of another project

Yes - that's why there's a license file in there - they come from the `rqlite` project: [db.go](https://github.com/rqlite/rqlite/blob/master/db/db.go) and its dependencies.

The changes I made were:
* fix warnings
* add context parameter to Open, Query & Execute in order to pass a context parameter and call functions in the `sql` package that use it, e.g. `ExecContext` or `QueryContext`.

**Pull request merged**